### ISSUE TITLE
Searcher UX: Handle arrow keys

### ISSUE THREAD

**Issue by with-heart** [mattpocock/xstate-catalogue#26]
Right now the searcher can only be navigated using tab, which feels unexpected for that type of interaction. Most sites with an Alogolia UI or similar support arrow keys.

Obviously not the highest priority, but it's a nice bit of polish that would make search feel extra special. Plus this would probably add enough unique functionality to make the searcher machine a good one to include in the catalogue itself, while also driving the search experience. That's my kind of dogfooding!

| Key | Cond | Action|
| --- | --- | --- |
| Down Arrow | Final item is focused | Move focus to first item |
| Down Arrow | - | Move focus to next item |
| Up Arrow | First item is focused | Move focus to last item |
| Up Arrow | - | Move focus to previous item |

**Comment by hmaurer** [comment]
👍 . The right way to do this would be to use a combobox primitive I think, e.g. https://reach.tech/combobox or https://www.downshift-js.com/use-combobox.

**Comment by mattpocock** [comment]
Yeah! Once the combobox state machine is written we can practice what we preach.
